scripts/install.py

```diff
@@ -1012,6 +1012,57 @@ def tweak_malcolm_runtime(self, malcolm_install_path):
+            loopBreaker = CountUntilException(
+                MaxAskForValueCount,
+                f'Invalid ILM setting(s)',
+            )
+            ilmPolicy =  InstallerYesOrNo(
+                    f'Should Arkime leverage Index Lifecycle Management (ILM) to manage volume?',
+                    default=args.ilmPolicy
+            )
+            if ilmPolicy:
+                while loopBreaker.increment():
+                    # Set 'hot' for 'node.attr.molochtype' on new indices, warm on non sessions indices
+                    ilmHotWarm =  InstallerYesOrNo(
+                        f'Should Arkime use a hot/warm design in which non-session data is stored in a warm index?',
+                        default=args.ilmHotWarm
+                    )
+                    if ilmHotWarm:
+                        InstallerDisplayMessage(
+                            f'You must configure "hot" and "warm" nodes types in the remote Elasticsearch instance (https://arkime.com/faq#ilm)'
+                        )
+                    # Time in hours/days before (moving Arkime indexes to warm) and force merge (number followed by h or d), default 30d
+                    ilmOptimizationTimePeriod = InstallerAskForString(
+                        "How long should Arkime keep an index in the hot node? (e.g. 25h, 5d, etc.)",
+                        default=args.ilmOptimizationTimePeriod
+                    )
+                    # Time in hours/days before deleting Arkime indexes (number followed by h or d), default 90d
+                    ilmSpiDataRetention = InstallerAskForString(
+                        "How long should Arkime retain SPI data before deleting it? (e.g. 25h, 90d, etc.)",
+                        default=str(args.ilmSpiDataRetention)
+                    )
+                    # Number of segments to optimize sessions to in the ILM policy, default 1
+                    ilmOptimizeSessionSegments = InstallerAskForString(
+                        "How many segments should Arkime use to optimize?",
+                        default=str(args.ilmOptimizeSessionSegments) 
+                    )
+                    # Number of replicas for older sessions indices in the ILM policy, default 0
+                    ilmReplicas = InstallerAskForString(
+                        "How many replicas should Arkime maintain for older session indices?",
+                        default=str(args.ilmReplicas)
+                    )
+                    # Number of weeks of history to keep, default 13
+                    ilmHistoryInWeeks = InstallerAskForString(
+                        "How many weeks of history should Arkime keep?",
+                        default=str(args.ilmHistoryInWeeks)
+                    )
+                    if (len(ilmOptimizationTimePeriod) > 1) and (len(ilmSpiDataRetention) > 1) and str(ilmOptimizeSessionSegments).isdigit() and str(ilmReplicas).isdigit() and str(ilmHistoryInWeeks).isdigit():
```

Comment:
If the only valid values for `ilmOptimizationTimePeriod` and `ilmSpiDataRetention` is a number followed by `h` or `d` you should do more thorough checking than just `len() > 1`. Use a regex (e.g., `^\d+[hd]$`) to validate these.